### PR TITLE
Update readme and remove certificates

### PR DESCRIPTION
**Pull request by HedicGuibert**
After struggling a bit running the project as a beginner, I thought it may be a good idea to update the readme to make it more beginner-friendly. So, this PR aims at giving more precise informations about what one should do to be able to run the project.

Also, I removed the provided key and certificate and added the **/infrastructure/docker/services/router/etc/ssl/certs/** directory to .gitgnore so that people won't be confused about what to do with them, and won't be able to commit them either.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,29 +7,56 @@
...
-Use [mkcert](https://github.com/FiloSottile/mkcert) to install a CA on your system to be able to generate a valid self-signed certificate for local.starfleet.app
+Then paste in `local.starfleet.app localhost 127.0.0.1`
 
-`$ mkcert *.starfleet.app`
+You now need to generate a certificate for local.starfleet.app. This is why you need mkcert, which you may find [here](https://github.com/FiloSottile/mkcert) with the installation instructions.
 
-> certificate and key must be placed in infrastructure/docker/services/router/etc/ssl/certs
+When mkcert is installed, go to your project directory and generate your certificate with :
+```bash
+mkcert local.starfleet.app "*.starfleet.app" local.starfleet.app localhost 127.0.0.1 ::1
```
Comment (welcoMattic) [inline README.md:34]: localhost is not mandatory here, and local.starfleet.app is written twice

Comment (welcoMattic) [reply README.md:34]: "*.starfleet.app" is suffisant

File: README.md [hunk 2]
```diff
@@ -7,29 +7,56 @@
...
+```
 
-Install Python virtualenv to be able to use fabric commands to control easily the Docker stack
+You will find 2 certificates at the root of your project with the following names :
+> "name-of-your-project.com+5.pem'
+> "name-of-your-project.com+5-key.pem'
 
-`$ pipenv install`
+Rename the one with the "-key" suffix to `key.pem` and the other one to `cert.pem`. Then, move them both to :
```
Comment (welcoMattic) [inline README.md:41]: mkcert let's you define the name of output files

File: README.md [hunk 3]
```diff
@@ -7,29 +7,56 @@
 
 > Share your conferences activity to your buddies
 
+## Requirements
+
+This project requires :
+
+    - Docker 
+    - mkcert (https://github.com/FiloSottile/mkcert)
+    - pipenv (https://github.com/pypa/pipenv)
```
Comment (lyrixx) [inline README.md:16]: Indentation seems wrong

File: README.md [hunk 4]
```diff
@@ -7,29 +7,56 @@
...
+
 ## Installation
 
 ### Local
 
-> Add `local.starfleet.app` to your hosts file
+First, you will need to add local.starfleet.app to your hosts file.
```
Comment (lyrixx) [inline README.md:22]: please, add backquote around all technical content. Like `local.starfleet.app`

File: README.md [hunk 5]
```diff
@@ -7,29 +7,56 @@
...
+Open it with :
+```
+sudo nano /etc/hosts
```
Comment (lyrixx) [inline README.md:25]: We does not need this. Be instead we need:
```
127.0.0.1 local.starfleet.app
```

File: README.md [hunk 6]
```diff
@@ -7,29 +7,56 @@
...
+```
 
-Use [mkcert](https://github.com/FiloSottile/mkcert) to install a CA on your system to be able to generate a valid self-signed certificate for local.starfleet.app
+Then paste in `local.starfleet.app localhost 127.0.0.1`
 
-`$ mkcert *.starfleet.app`
+You now need to generate a certificate for local.starfleet.app. This is why you need mkcert, which you may find [here](https://github.com/FiloSottile/mkcert) with the installation instructions.
```
Comment (lyrixx) [inline README.md:30]: you can replace "wich you may find here ..." by a link on "mkcert" directly.

File: README.md [hunk 7]
```diff
@@ -7,29 +7,56 @@
...
+```
 
-Install Python virtualenv to be able to use fabric commands to control easily the Docker stack
+You will find 2 certificates at the root of your project with the following names :
+> "cert.pem'
+> "key.pem'
 
-`$ pipenv install`
+Move them both to :
+`./infrastructure/docker/services/router/etc/ssl/certs/`
```
Comment (lyrixx) [inline README.md:43]: Could be replaced with options provided by mkcert.

BTW, I would use inv to do that. So the readme would become

```
inv generate-certificate
```

File: README.md [hunk 8]
```diff
@@ -7,29 +7,56 @@
...
 
-Enter in pipenv shell to get access to `inv` command
+Now, you need to install Python virtualenv to be able to use fabric commands to easily control the Docker stack. See instructions [here](https://github.com/pypa/pipenv.).
```
Comment (lyrixx) [inline README.md:44]: Same about the link "here". It's a bad practice to name a link "here" BTW

File: README.md [hunk 9]
```diff
@@ -7,29 +7,54 @@
...
+When mkcert is installed, go to your project directory and generate your certificate with :
+```bash
+mkcert -cert-file ./infrastructure/docker/services/router/etc/ssl/certs/cert.pem -key-file ./infrastructure/docker/services/router/etc/ssl/certs/key.pem "*.starfleet.app"
+```
 
-Install Python virtualenv to be able to use fabric commands to control easily the Docker stack
+Or you could also do it with invoke. Wait till you ran `pipenv shell` (see the step below) and then run
+```bash
+inv generate-certificate
+```
```
Comment (lyrixx) [inline README.md:40]: The command does not exist.

File: README.md [hunk 10]
```diff
@@ -7,29 +7,54 @@
...
+Then paste in `127.0.0.1 local.starfleet.app`
 
-`$ mkcert *.starfleet.app`
+You now need to generate a certificate for `local.starfleet.app`. This is why you need mkcert, which you may find here with the installation instructions : https://github.com/FiloSottile/mkcert
 
-> certificate and key must be placed in infrastructure/docker/services/router/etc/ssl/certs
+When mkcert is installed, go to your project directory and generate your certificate with :
+```bash
+mkcert -cert-file ./infrastructure/docker/services/router/etc/ssl/certs/cert.pem -key-file ./infrastructure/docker/services/router/etc/ssl/certs/key.pem "*.starfleet.app"
+```
```
Comment (lyrixx) [inline README.md:35]: It's useless to propose two way to do the same thing. Keep only the last one. Thanks

File: README.md [hunk 11]
```diff
@@ -7,29 +7,54 @@
...
+Open it with :
+```
+sudo nano /etc/hosts
+```
 
-Use [mkcert](https://github.com/FiloSottile/mkcert) to install a CA on your system to be able to generate a valid self-signed certificate for local.starfleet.app
+Then paste in `127.0.0.1 local.starfleet.app`
 
-`$ mkcert *.starfleet.app`
+You now need to generate a certificate for `local.starfleet.app`. This is why you need mkcert, which you may find here with the installation instructions : https://github.com/FiloSottile/mkcert
```
Comment (lyrixx) [inline README.md:30]: 1. There are never space before double colon in English.
2. The sentence if too long IHMO

```
You need [mkcert](https://github.com/FiloSottile/mkcert) to generate a certificate for `local.starfleet.app`:
```

File: README.md [hunk 12]
```diff
@@ -7,29 +7,54 @@
 
 > Share your conferences activity to your buddies
 
+## Requirements
+
+This project requires :
+
+- Docker 
+- mkcert (https://github.com/FiloSottile/mkcert)
+- pipenv (https://github.com/pypa/pipenv)
```
Comment (lyrixx) [inline README.md:16]: Put link directly on the tools

File: README.md [hunk 13]
```diff
@@ -25,23 +25,21 @@ Open it with :
...
-
-Move them both to :
-`./infrastructure/docker/services/router/etc/ssl/certs/`
+Or you could also do it with invoke. Wait till you ran `pipenv shell` (see the step below) and then run
+```bash
+inv generate-certificate
+```
 
-Now, you need to install Python virtualenv to be able to use fabric commands to easily control the Docker stack. See instructions [here](https://github.com/pypa/pipenv.).
+Now, you need to install Python virtualenv to be able to use fabric commands to easily control the Docker stack. See instructions : https://github.com/pypa/pipenv.
```
Comment (welcoMattic) [inline README.md:42]:
```suggestion
Now, you need to install Python [pipenv](https://github.com/pypa/pipenv) to be able to use fabric commands to easily control the Docker stack.
```

File: README.md [hunk 14]
```diff
@@ -7,32 +7,59 @@
...
+```
 
-`$ mkcert *.starfleet.app`
+Then paste in `127.0.0.1 local.starfleet.app`
 
-> certificate and key must be placed in infrastructure/docker/services/router/etc/ssl/certs
+You then need [mkcert](https://github.com/FiloSottile/mkcert) to generate a certificate for `local.starfleet.app`
 
-Install Python virtualenv to be able to use fabric commands to control easily the Docker stack
+When mkcert is installed, you need to install Python [pipenv](https://github.com/pypa/pipenv) to be able to use fabric commands to easily control the Docker stack.
```
Comment (lyrixx) [inline README.md:32]: The workflow is a bit odd. You talk about mkdir, but then the command is about pipenv, 
So IMHO, you must put the mkcert sentence after the pipenv block

File: README.md [hunk 15]
```diff
@@ -7,32 +7,59 @@
...
+```
 
 Open [https://local.starfleet.app](https://local.starfleet.app) in your browser 🚀
 
+You may then need to update the project packages. For this, run
+```bash
+inv builder
+```
+
+You may now update composer with `composer u` !
```
Comment (lyrixx) [inline README.md:61]: Nope!, to remove

File: tasks.py [hunk 16]
```diff
@@ -281,6 +281,14 @@ def help(c):
     except:
         pass
 
+@task
+def generate_certificates(c):
+    """
+    Generates the cert.pem and cert-key.pem files
+    """
+    with c.cd(c.project_directory):
+        c.run('mkcert -cert-file ./infrastructure/docker/services/router/etc/ssl/certs/cert.pem -key-file ./infrastructure/docker/services/router/etc/ssl/certs/key.pem "*.starfleet.app"')
```
Comment (lyrixx) [inline tasks.py:290]: Lets go the the right directory straight away, it will avoid duplucation of path

File: README.md [hunk 17]
```diff
@@ -7,32 +7,52 @@
...
 
-`$ mkcert *.starfleet.app`
+### Local
 
-> certificate and key must be placed in infrastructure/docker/services/router/etc/ssl/certs
+First, you will need to add `local.starfleet.app` to your hosts file.
+> `127.0.0.1 local.starfleet.app`
 
-Install Python virtualenv to be able to use fabric commands to control easily the Docker stack
+Install Python [pipenv](https://github.com/pypa/pipenv) to be able to use fabric commands to easily control the Docker stack.
```
Comment (lyrixx) [inline README.md:27]: I re-read the whole README, and IMHO, we don't need to talk about python pipenv here, since we have already talked about it in the `Requirements` section.

TL;DR: You can simply remove this sentence.

File: README.md [hunk 18]
```diff
@@ -7,32 +7,52 @@
...
 
-`$ pipenv install`
+You can then run
```
Comment (lyrixx) [inline README.md]: This sentence can be remove too, or it could be replace by "Install pipenv dependencies"

File: README.md [hunk 19]
```diff
@@ -7,32 +7,52 @@
...
+pipenv install
+```
 
 Enter in pipenv shell to get access to `inv` command
+```bash
+pipenv shell
+```
 
-`$ pipenv shell`
+You then need [mkcert](https://github.com/FiloSottile/mkcert) to generate a certificate for `local.starfleet.app`
```
Comment (lyrixx) [inline README.md:39]: Same comment as pipenv. You can replace this sentence with the following by:

```
Generate SSL certificates:
```

File: README.md [hunk 20]
```diff
@@ -7,32 +7,54 @@
...
 
-`$ inv start`
+If it's your first install of Starfleet, run:
+```bash
+inv start
+```
 
 Open [https://local.starfleet.app](https://local.starfleet.app) in your browser 🚀
 
+You may then need to update the project packages. For this, run:
```
Comment (lyrixx) [inline README.md:53]: The builder is not only for update project package. I would reword this to

```
If you need to enter a shell to run specific command, run the following command:
```

File: tasks.py [hunk 21]
```diff
@@ -281,6 +281,14 @@ def help(c):
     except:
         pass
 
+@task
+def generate_certificates(c):
+    """
+    Generates the cert.pem and cert-key.pem files
+    """
+    with c.cd(c.project_directory + 'infrastructure/docker/services/router/etc/ssl/certs/'):
```
Comment (lyrixx) [inline tasks.py:289]: I did not test it, but I think it does not work. It misses a leading `/` before infrastructure.
I say that, because It's a best practice to never put trailing `/` in a variable container a path

Comment (welcoMattic) [inline tasks.py:289]: We need to be sure that the directory exists, and if not, to create it

Comment (HedicGuibert) [reply tasks.py]: I added a .gitignore to the certs directory to make sure it exists but I can add this verification if you'd like it

**Review by lyrixx — APPROVED** [review]
Nice, looks good to me 👍🏼

**Comment by lyrixx** [comment]
Good catch, thanks @HedicGuibert.
